### PR TITLE
Remove 'Checking compatibility...' log

### LINKED ISSUE #10383: Remove "Checking compatibility..." log messages from RestoreTask
Currently RestoreTask logs a lot of messages that start with Checking compatibility:

image

The log output is full of these. I'm questioning the value of having these in the logs since they don't provide much useful information and generate a ton of noise.

I suggest that we just stop logging this particular message.

### PR DESCRIPTION
**Pull request by Nigusu-Allehu**
## Bug

Fixes: https://github.com/NuGet/Home/issues/10383

Regression? Last working version:

## Description

Remove "Checking compatibility ..." logs. It's agreed this log does not provide any useful information. 
## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [x] Test exception : I run `msbuild /t:restore [solution path] /v:d >log.txt` before and after the changes. The logs were removed after the changes.
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by jeffkl — DISMISSED** [review]
It's fascinating that there are no unit tests to update/remove as well.

**Comment by aortiz-msft** [comment]
Please fill out the PR description and under manual tests, describe what manual testing you performed to validate this change.

**Review by aortiz-msft — CHANGES_REQUESTED** [review]
Please fill out the PR description and under manual tests, describe what manual testing you performed to validate this change.

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs [hunk 1]
```diff
@@ -1062,7 +1062,6 @@ internal static Task LogDowngradeWarningsOrErrorsAsync(IEnumerable<RestoreTarget
                     // Don't do compat checks for the ridless graph of DotnetTooReference restore. Everything relevant will be caught in the graph with the rid
                     if (!(ProjectStyle.DotnetToolReference == project.RestoreMetadata?.ProjectStyle && string.IsNullOrEmpty(graph.RuntimeIdentifier)))
                     {
-                        await logger.LogAsync(LogLevel.Verbose, string.Format(CultureInfo.CurrentCulture, Strings.Log_CheckingCompatibility, graph.Name));
```
Comment (aortiz-msft) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs]: Could you please remove the empty line after the {?

**Review by aortiz-msft — DISMISSED** [review]

**Review by nkolev92 — APPROVED** [review]

**Review by aortiz-msft — APPROVED** [review]

**Pull request merged**